chore: Add script to ease local development between veda-ui and next-veda-ui

dzole0311 opened the pull request:
**Related Ticket:** https://github.com/NASA-IMPACT/veda-ui/issues/1549

### Description of Changes
Adds a `link-next.sh` script to make local linking and development between veda-ui into next-veda-ui easier. The script builds the veda-ui library, links it globally, links it into the Next.js project and watches for changes in veda-ui to re-trigger `yarn buildlib`.

### Notes & Questions About Changes

1. Assumes both repos are in sibling directories
2. One caveat is that during rebuilds, you may briefly see a `Module not found: Can't resolve '/veda-ui/lib/main.css'` error in Next.js. This happens because each new `yarn buildlib` clears the previous build. The error will resolve once the rebuild completes

### Validation / Testing

Make sure you run `yarn` to install the new dependency added in the project.

1. Open a terminal and run the script `./utils/link-next.sh` in the veda-ui directory
3. Once it's done, open another terminal and run the next-veda-ui project (`npm run dev` in the next-veda-ui directory)
4. Make sure you're using the main branch in the next-veda-ui project for this test
5. Open `http://localhost:3000/` in your browser
6. Change something in a veda-ui component used in Next.js (e.g. add an extra className 'test' to the `usa-navbar`)
7. Verify that the library is rebuilt and the change is visible in Next.js

sandrahoang686 commented:
Thanks for working on this! We could seriously benefit a ton from this...

I ran through the steps but I encountered some errors 😢 . I followed your instructions above and when it come to making a change in the veda-ui component, i ran into this error below:
![Screenshot 2025-04-02 at 5 19 15 PM](https://github.com/user-attachments/assets/ee796a09-6e36-4827-bd1f-d5247e522935)

And i'm wondering does it have to do with the recent updates made to the entrypoints that Hanbyul made [here](https://github.com/NASA-IMPACT/veda-ui/pull/1535/files). 🤔 very unsure.

So i exited the process and then for some reason decided to do a `yarn install` in next-veda-ui and then re-ran through your steps where now I couldn't even get the nextJs app to run because I ran into this error that I commonly see...
``` ⨯ ../veda-ui/node_modules/react-router/dist/index.js:12:1
Module not found: Can't resolve '@remix-run/router'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
../veda-ui/lib/module.js
./app/lib/index.ts
 GET /data-catalog 500 in 27912ms```
 
 Any ideas?? @snmln have you encountered the same? or is it just me...?

sandrahoang686 reviewed (approved):
my comments could be follow-ups. Approving since this works for me and is already an improvement. would be nice if someone else could walk through this too though.

dzole0311 commented:
Thanks @sandrahoang686 , I've updated the script to also unlink previously linked packages (if any). Also, I updated our "REGISTRY.md" doc with these setup: https://github.com/NASA-IMPACT/veda-ui/pull/1575/files#diff-f28310bf68e9852de9f79f8f8e7f26c1e1948dbccc4102223b1533b46e05d33e
